titus-common-runtime/src/main/java/com/netflix/titus/runtime/connector/kubernetes/KubeApiClients.java

```diff
@@ -76,6 +88,24 @@ public static ApiClient createApiClient(String kubeApiServerUrl,
         return client;
     }
 
+    public static Either<Boolean, Throwable> checkKubeConnectivity(ApiClient apiClient) {
+        CoreV1Api coreV1Api = new CoreV1Api(apiClient);
+        try {
+            coreV1Api.getAPIResources();
+        } catch (Throwable e) {
+            return Either.ofError(e);
+        }
+        return Either.ofValue(true);
+    }
+
+    public static ApiClient mustHaveKubeConnectivity(ApiClient apiClient) {
+        Either<Boolean, Throwable> check = checkKubeConnectivity(apiClient);
```

Comment:
not checking the boolean return anywhere

Reply:
The Either contract in this context makes it sufficient to only check if the error is present.